applications/GeoMechanicsApplication/custom_strategies/strategies/residualbased_newton_raphson_strategy_linear_elastic_dynamic.hpp

```diff
@@ -0,0 +1,810 @@
+        Parameters&                                rParameters,
+        double                                     Beta,
+        double                                     Gamma,
+        int                                        MaxIterations      = 30,
+        bool                                       CalculateReactions = false,
+        bool                                       MoveMeshFlag       = false)
+        : ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(
+              rModelPart, pScheme, pNewConvergenceCriteria, pNewBuilderAndSolver, MaxIterations, CalculateReactions, false, MoveMeshFlag),
+          mBeta(Beta),
+          mGamma(Gamma)
+    {
+        // new constructor
+    }
+
+    /**
+     * @brief Operation to predict the solution
+     */
+    void Predict() override
+    {
+        KRATOS_TRY
+        // OPERATIONS THAT SHOULD BE DONE ONCE - internal check to avoid repetitions
+        // if the operations needed were already performed this does nothing
+        if (BaseType::mInitializeWasPerformed == false) Initialize();
+
+        const auto equation_size = BaseType::GetBuilderAndSolver()->GetDofSet().size();
+
+        ModelPart& model_part = BaseType::GetModelPart();
+
+        double delta_time = model_part.GetProcessInfo()[DELTA_TIME];
+
+        TSystemVectorType first_derivative_vector  = TSystemVectorType(equation_size, 0.0);
+        TSystemVectorType second_derivative_vector = TSystemVectorType(equation_size, 0.0);
+
+        TSystemVectorType updated_first_derivative_vector  = TSystemVectorType(equation_size, 0.0);
+        TSystemVectorType updated_second_derivative_vector = TSystemVectorType(equation_size, 0.0);
+
+        this->GetFirstAndSecondDerivativeVector(first_derivative_vector, second_derivative_vector,
+                                                model_part, 0);
+
+        // performs: updated_first_derivative_vector = first_derivative_vector * (mGamma / mBeta) + second_derivative_vector * (delta_time * (mGamma / (2 * mBeta) - 1));
+        TSparseSpace::ScaleAndAdd(mGamma / mBeta, first_derivative_vector,
+                                  delta_time * (mGamma / (2 * mBeta) - 1), second_derivative_vector,
+                                  updated_first_derivative_vector);
+
+        // performs: updated_second_derivative_vector = first_derivative_vector * (1.0 / (mBeta * delta_time)) + second_derivative_vector * (1.0 / (2 * mBeta));
+        TSparseSpace::ScaleAndAdd(1.0 / (mBeta * delta_time), first_derivative_vector, 1.0 / (2 * mBeta),
+                                  second_derivative_vector, updated_second_derivative_vector);
+
+        this->SetFirstAndSecondDerivativeVector(updated_first_derivative_vector,
+                                                updated_second_derivative_vector, model_part);
+
+        // Move the mesh if needed
+        if (BaseType::MoveMeshFlag()) BaseType::MoveMesh();
```

Comment:
we've never used in the geomechanics application, i do not know what it actually does, and i do not want to add unused/untested stuff

Reply:
We actually do now for a practical model including a sheet pile installation. Since the only reason we cannot use this predict now seems to be that the formulas of the scheme don't align, I'd propose to align the formulas and re-use the base Predict. That way we still include the master slave constraints and they are tested via the base class.